nameserver/nameserver.go

```diff
@@ -121,20 +252,78 @@ func (n *Nameserver) ReverseLookup(ip address.Address) (string, error) {
 	return match.Hostname, nil
 }
 
-func (n *Nameserver) ContainerStarted(ident string)   {}
-func (n *Nameserver) ContainerDestroyed(ident string) {}
+func (n *Nameserver) ContainerStarted(ident string) {}
+
+func (n *Nameserver) ContainerDestroyed(ident string) {
+	n.Lock()
+	if !n.ready {
+		n.pendingEvents = append(n.pendingEvents, event{eventDestroyed, ident})
+		n.Unlock()
+		return
+	}
+	entries := n.containerDestroyed(ident)
+	n.Unlock()
+	n.broadcastEntries(entries...)
+}
 
 func (n *Nameserver) ContainerDied(ident string) {
 	n.Lock()
+	if !n.ready {
+		n.pendingEvents = append(n.pendingEvents, event{eventDied, ident})
+		n.Unlock()
+		return
+	}
+	entries := n.containerDied(ident)
+	n.Unlock()
+	n.broadcastEntries(entries...)
+}
+
+func (n *Nameserver) containerDestroyed(ident string) Entries {
+	entries := n.entries.forceTombstone(n.ourName, func(e *Entry) bool {
```

Comment:
why do we need `forceTombstone`?  You don't seem to be changing `e` in the case where you return `false`.